packages/api-core/src/api.js

```diff
@@ -118,15 +119,26 @@ export default class AvApi {
 
   // return location if should poll otherwise false
   getLocation(response) {
-    let location = false;
+    let locationUrl;
+    const { config, headers = {} } = response;
+    const { getHeader, base, url } = config;
+    const { location, Location } = headers;
 
-    if (response.config.getHeader) {
-      location = response.config.getHeader(response, 'Location');
+    if (getHeader) {
+      locationUrl = getHeader(response, 'Location');
     } else {
-      location = response.headers.location || response.headers.Location;
+      locationUrl = location || Location;
     }
 
-    return resolveUrl(location);
+    if (locationUrl && !isAbsoluteUrl(locationUrl)) {
+      if (base) {
+        locationUrl = resolveUrl({ relative: location, base });
+      } else if (url && isAbsoluteUrl(url)) {
+        locationUrl = resolveUrl({ relative: location, url });
```

Comment:
I think you wanted `locationUrl` instead of `location` when making the object to pass to `resolveUrl` here and 2 lines above.
Also, the key for the base is `base`, when using the short-hand here, it's going to be `url` which doesn't looks like the function is expecting.

Not 100% sure, but I think this whole nested if can be reduced since `resolveUrl` has a built-in `isAbsoluteUrl` which just returns the url if it's already absolute.

```js
return resolveUrl({ relative: locationUrl, base: base || url });
```

Reply:
much cleaner 🙏